### DISCUSSION TITLE
Github copilot 1.1.35.2063 code completion doesn't work

### DISCUSSION THREAD

**Discussion by stephen-sgray-net** [community/community#39726] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi,
I am using copilot latest version in intellij 2021.3.3. When I type some code in an editor and hit refresh in copilot window it says "Retrieving completions..." and hangs. It never comes back with a completion. I am behind a proxy and have my proxy settings setup in intellij. Can someone advise? As it is it is unusable.
Thanks,
Steve

**Comment by stephen-sgray-net** [comment]
I have found something in the intellij log (idea.log):
2022-11-23 13:37:51,508 [3445009]   INFO -                       #copilot - [INFO] [default] [2022-11-23T02:37:51.502Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex 
2022-11-23 13:37:51,946 [3445447]   WARN -                       #copilot - [ERROR] [default] [2022-11-23T02:37:51.945Z] Request Error: unable to verify the first certificate 
2022-11-23 13:37:51,946 [3445447]   WARN -                       #copilot - [ERROR] [ghostText] [2022-11-23T02:37:51.947Z] Error on ghost text request FetchError: unable to verify the first certificate 

Obviously copilot is trying to access its own site via https and failing. But I have set up https proxy information under intellij settings, and check connection to https://www.google.com succeeds. Can anyone help?

**Comment by ossanna16** [comment]
Hi there @stephen-sgray-net and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by stephen-sgray-net** [comment, marked as answer]
OK, for anyone else who has this issue. You need to trust the certificate provided by the website you are making your SSL call. The usual way of doing this in intellij is through the Settings/Tools/Server Certificates dialog, which allows you to import certificates that are "accepted" (trusted). Unfortunately copilot ignores this list, and also ignores certificates imported into the JRE certificate keystore (<intellij install folder>/jbr/lib/security/cacerts). It seems that copilot uses nodejs code to request. You can tell nodesj not to verify certificates by setting a windows environment variable NODE_TLS_REJECT_UNAUTHORIZED to 0 (see https://stackoverflow.com/a/67842168). WARNING: this will disable all certificate verification for nodejs calls. 

The best solution would be for copilot to check the Server Certificates accepted certificates in intellij when making SSL calls (are you listening copilot developers?).

Thanks to the helpful intellij tech support for the answer to this problem.

**Comment by sittingbool** [comment]
I am having this issue again on WebStorm after the recent update. I had it resolved for a day now by restarting the IDE

**Reply by sittingbool** [reply]
I did the last two updates. It still does that after a while, then I always try to shut down weStorm and end up having to kill the process because its not even shutting down anymore. It's becoming real frustrating. It also slows down my computer like hell.
Happens after coding about 2-3 hours.
